### PR TITLE
SDK - Tests - Improved tests for serializing lists containing objects

### PR DESCRIPTION
**Pull request by Ark-kun**
Added test_fail_on_handling_list_arguments_containing_python_objects
Added test_handling_list_arguments_containing_serializable_python_objects
Moved test_handling_list_arguments_containing_pipelineparam to component_bridge_tests

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3326)

### CODE REVIEW

File: sdk/python/tests/dsl/component_bridge_tests.py [hunk 1]
```diff
@@ -181,3 +181,16 @@ def calc_pipeline(
...
             kfp.compiler.Compiler().compile(calc_pipeline, pipeline_filename)
+
+    def test_handling_list_arguments_containing_pipelineparam(self):
+        '''Checks that lists containing PipelineParam can be properly serialized'''
+        def consume_list(list_param: list) -> int:
+            pass
+
+        import kfp
+        task_factory = create_component_from_func(consume_list)
+        task = task_factory([1, 2, 3, kfp.dsl.PipelineParam("aaa"), 4, 5, 6])
```
Comment (numerology) [inline sdk/python/tests/dsl/component_bridge_tests.py]:
```suggestion
        task = task_factory([1, 2, 3, kfp.dsl.PipelineParam('aaa'), 4, 5, 6])
```

Comment (Ark-kun) [reply sdk/python/tests/dsl/component_bridge_tests.py]: Done

**Pull request merged**